src/udx.c

```diff
@@ -417,8 +430,23 @@ process_packet (udx_t *socket, char *buf, ssize_t buf_len) {
   buf += UDX_HEADER_SIZE;
   buf_len -= UDX_HEADER_SIZE;
 
-  udx_stream_t *stream = (udx_stream_t *) udx__cirbuf_get(&(socket->streams_by_id), local_id);
-  if (stream == NULL || stream->status & UDX_STREAM_DEAD) return 0;
+  udx_stream_t *stream = lookup_stream(socket, local_id);
+
+  if (stream == NULL || (stream->status & UDX_STREAM_CONNECTED) == 0) {
+    if (socket->on_preconnect != NULL) {
```

Comment:
Can we add a comment here saying the preconnect hook is to give users a chance to connect the stream if it's not connected, hence we do the logic twice here